pkg/clustermesh/operator/remote_cluster.go

```diff
@@ -89,6 +89,10 @@ func (rc *remoteCluster) Stop() {
 	rc.synced.Stop()
 }
 
+func (rc *remoteCluster) RevokeCache(ctx context.Context) {
+	rc.Remove(ctx)
```

Comment:
Even if it's currently roughly the same thing I would prefer that this calls `Drain` here so that we would not be surprised if we ever watch more resource on the operator and get confused of this difference wrt the agent:

```suggestion
	rc.remoteServices.Drain()
	rc.remoteServiceExports.Drain()
```